### ISSUE TITLE
Remote management control  of coredns

### ISSUE THREAD

**Issue by nsinghisking** [coredns/coredns#1182]
Other popular dnsservers like bind/unbound provide ability to control and get information/statistics from the running instance of dns server. It will be nice to have a similar functionality in coredns server as well.

Few of the necessary things will be.

- DNS Statistics
-  List any local Zones.

**Comment by miekg** [comment]
* The [prometheus](https://coredns.io/plugins/metrics) plugin should give you enough statistics, and if not please file a specific bug for that.

Other than the [auto](https://coredns.io/plugins/auto) plugin, no other plugin picks up a new zone automatically, meaning the Corefile relfects what is loaded. [catalog zones](https://tools.ietf.org/html/draft-muks-dnsop-dns-catalog-zones-03) might be nice though.

Think the question I have boils down to why do you think you need this?

**Comment by fturib** [comment]
@nsinghsking : what are the 'control' operation you would expect on the running instance ?

**Comment by nsinghisking** [comment]
@miekg I will try prometheus plugin and see if it provides all options. I am mostly looking for numbers of queries received/dropped/responded, Cache hits and QPS rate
@fturib I was looking for few options as cache dump, flush cache

**Comment by miekg** [comment]
Cache flush could be done by just reloading the process via SIGUSR1.
Dumping the cache contents  isn't possible and I agree that would be nice, but creating a whole new interface with authn, authz and TLS is a bit much.

**Comment by miekg** [comment]
Alternatively we *could* add something like the *pprof* plugin, that has a web endpoint (by default on localhost) that just dumps this as read-only info.

Something like this would help in debugging issues.

**Comment by miekg** [comment]
this will not happen any time soon.

**Comment by frittentheke** [comment]
> Other than the [auto](https://coredns.io/plugins/auto) plugin, no other plugin picks up a new zone automatically, meaning the Corefile relfects what is loaded. [catalog zones](https://tools.ietf.org/html/draft-muks-dnsop-dns-catalog-zones-03) might be nice though.

@miekg I was just about to open an issue about adding support for reading a zone list from catalog zones to have CoreDNS serve them as secondary. I just proposed adding the ability to provide catalog zones to OpenStack Designate (https://blueprints.launchpad.net/designate/+spec/catalog-zones). Designate is also topic of another issue (https://github.com/coredns/coredns/issues/5069) in regards to supporting CoreDNS as a backend.

It seems so obvious to have AXFR for zones and also their records as "the standard" way of provisioning a secondary DNS into CoreDNS, apart from adding more and more plugins for particular data sources.

In short: Would you consider a plugin or extension of the existing _auto_ plugin to support reading catalog_zones to provision secondary zones into CoreDNS?

**Comment by chrisohaver** [comment]
@frittentheke, please open a new enhancement request for discussion.  There we can discuss how it could be implemented, e.g. as new plugin, or by extending the _auto_ plugin, or extending the _secondary_ plugin, etc
